bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,542 @@
+                
+        public CBusThreadPoolExecutor(@Nullable String poolName)
+        {
+            threadPool = (ThreadPoolExecutor) ThreadPoolManager.getPool(poolName);
+        }
+        
+        @Override
+        protected void execute(@Nullable Runnable runnable)
+        {
+            threadPool.execute(runnable);
+        }
+    }
+    public CBusThreadPool()
+    {}
+    private Map<@Nullable String, CGateThreadPoolExecutor> executorMap= new HashMap<@Nullable String, CGateThreadPoolExecutor>();
+
+    @Override
+    @SuppressWarnings({ "unused", "null" }) /* map.get() can return null */
+    protected  CGateThreadPoolExecutor CreateExecutor(@Nullable String name)
+    {
+        @Nullable CGateThreadPoolExecutor executor = executorMap.get(name);
+
+        if (executor != null)
+        {
+            return executor;
+        }
+        CGateThreadPoolExecutor newExecutor =  new CBusThreadPoolExecutor(name);
+        executorMap.put(name, newExecutor);
+        return newExecutor;
+    }
+}
+
+/**
+ * The {@link CBusCGateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+
+@NonNullByDefault
+public class CBusCGateHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(CBusCGateHandler.class);
+
+    private @Nullable InetAddress ipAddress = null;
+
+    public @Nullable CGateSession cGateSession = null;
+
+    private @Nullable Future<?> keepAliveFuture = null; 
+
+    private final ExecutorService threadPool = ThreadPoolManager.getPool("CBusCGateHandler-Helper");
+
+    public CBusCGateHandler(Bridge br) {
+        super(br);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do here
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.OFFLINE);
+        logger.debug("Initializing CGate Bridge handler.");
+        Configuration config = getThing().getConfiguration();
+        String ipAddress = (String) config.get(CBusBindingConstants.PROPERTY_IP_ADDRESS);
+
+        if ("127.0.0.1".equals(ipAddress) || "localhost".equals(ipAddress)) {
+            this.ipAddress = InetAddress.getLoopbackAddress();
+        } else {
+            try {
+                this.ipAddress = InetAddress.getByName(ipAddress);
+            } catch (UnknownHostException e1) {
+                updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "IP Address not resolvable");
+                return;
+            }
+        }
+        InetAddress address = this.ipAddress;
+        if (address != null)
+            logger.debug("CGate IP         {}.", address.getHostAddress());
+
+        keepAliveFuture = threadPool.submit(new KeepAlive());
+    }
+
+    @NonNullByDefault
+    private class KeepAlive extends Thread {
```

Comment:
You define this as a thread but only use it as a Runnable. Please just change this to implement Runnable instead of extending Thread.

Reply:
Changed so this is now a Runnable.